src/map/mob.cpp

```diff
@@ -4981,9 +4930,9 @@ static void mob_drop_ratio_adjust(void){
 
 			// Item is not known anymore(should never happen)
 			if( !id ){
-				ShowWarning( "Monster \"%s\"(id:%d) is dropping an unknown item(id: %d)\n", mob->name, mob_id, nameid );
-				mob->dropitem[j].nameid = 0;
-				mob->dropitem[j].p = 0;
+				ShowWarning( "Monster \"%s\"(id:%hu) is dropping an unknown item(id: %d)\n", pair.second.name, pair.first, nameid );
+				pair.second.dropitem[j].nameid = 0;
+				pair.second.dropitem[j].p = 0;
```

Comment:
If you're using `.second` or `.first` like here more often, you could use references for better readability. Like `struct mob_db& mob = pair.second;`